src/createClient.spec.ts

```diff
@@ -130,28 +130,93 @@ describe('query', () => {
 })
 
 describe('useQuery', () => {
-  testInEffectScope('responds to parameter changes', async () => {
-    const responseTrue = Symbol('responseTrue')
-    const responseFalse = Symbol('responseFalse')
+  describe('when parameters change', () => {
+    testInEffectScope('response is updated', async () => {
+      const responseTrue = Symbol('responseTrue')
+      const responseFalse = Symbol('responseFalse')
 
-    const action = vi.fn((value: boolean) => value ? responseTrue : responseFalse)
-    const { useQuery } = createClient()
+      const action = vi.fn((value: boolean) => value ? responseTrue : responseFalse)
+      const { useQuery } = createClient()
 
-    const input = ref(false)
+      const input = ref(false)
 
-    const query = useQuery(action, () => [input.value])
+      const query = useQuery(action, () => [input.value])
 
-    expect(query.response).toBe(undefined)
+      expect(query.response).toBe(undefined)
 
-    await nextTick()
+      await nextTick()
 
-    expect(query.response).toBe(responseFalse)
+      expect(query.response).toBe(responseFalse)
 
-    input.value = true
+      input.value = true
 
-    await nextTick()
+      await nextTick()
 
-    expect(query.response).toBe(responseTrue)
+      expect(query.response).toBe(responseTrue)
+    })
+
+    testInEffectScope('executed and executing are updated', async () => {
```

Comment:
I wrote this test to prove that the object assign for a new query works for other `executed` and `executing` as well. 

knowing how it actually works, I think this makes sense but worth noting that when you change args the `executed` gets set to `false`, even if the "same query" (from users perspective) did execute before args changed

Reply:
Yeah I want to think through this carefully. Executed might be preserved when parameters change similar to response. But for now this is good